x-pack/plugins/observability_solution/observability_ai_assistant_app/public/functions/visualize_esql.tsx

```diff
@@ -238,88 +245,170 @@ export function VisualizeESQL({
+              </EuiText>
+              <EuiDescriptionList data-test-subj="observabilityAiAssistantErrorsList">
+                {errorMessages?.map((error, index) => {
+                  return (
+                    <EuiDescriptionListDescription key={index}>
+                      <EuiFlexGroup gutterSize="s" alignItems="center">
+                        <EuiFlexItem grow={false}>
+                          <EuiIcon type="error" color="danger" size="s" />
+                        </EuiFlexItem>
+                        <EuiFlexItem grow={false}>{error}</EuiFlexItem>
+                      </EuiFlexGroup>
+                    </EuiDescriptionListDescription>
+                  );
+                })}
+              </EuiDescriptionList>
+            </>
+          )}
+
+          <EuiFlexItem grow={false}>
+            <EuiFlexGroup direction="row" gutterSize="s" justifyContent="flexEnd">
+              {!isTableVisible && (
+                <EuiFlexItem grow={false}>
+                  <EuiToolTip
+                    content={i18n.translate(
+                      'xpack.observabilityAiAssistant.lensESQLFunction.table',
+                      {
+                        defaultMessage: 'Table of results',
+                      }
+                    )}
+                  >
+                    <EuiButtonIcon
+                      size="xs"
+                      iconType="tableDensityExpanded"
+                      onClick={() => setIsTableVisible(true)}
+                      data-test-subj="observabilityAiAssistantLensESQLDisplayTableButton"
+                      aria-label={i18n.translate(
+                        'xpack.observabilityAiAssistant.lensESQLFunction.displayTable',
+                        {
+                          defaultMessage: 'Display results',
+                        }
+                      )}
+                    />
+                  </EuiToolTip>
+                </EuiFlexItem>
+              )}
+              {isTableVisible && (
+                <EuiFlexItem grow={false}>
+                  <EuiToolTip
+                    content={i18n.translate(
+                      'xpack.observabilityAiAssistant.lensESQLFunction.visualization',
+                      {
+                        defaultMessage: 'Visualization',
+                      }
+                    )}
+                  >
+                    <EuiButtonIcon
+                      size="xs"
+                      iconType="visBarVerticalStacked"
+                      onClick={() => setIsTableVisible(false)}
+                      data-test-subj="observabilityAiAssistantLensESQLDisplayChartButton"
+                      aria-label={i18n.translate(
+                        'xpack.observabilityAiAssistant.lensESQLFunction.displayChart',
+                        {
+                          defaultMessage: 'Display chart',
+                        }
+                      )}
+                    />
+                  </EuiToolTip>
```

Comment:
seems like we can simplify this a bit? in both cases we render a EuiTooltip with a EuiButtonIcon

Reply:
Done https://github.com/elastic/kibana/pull/184106/commits/da816e8d9bf64e4a93a7ede44f078ff832598f04